text/logs/0114-log-correlation.md

```diff
@@ -0,0 +1,137 @@
+# Conventions for Trace and Resource Association in Logs
+
+To set out standards for correlating traces and resources in existing logs
+
+## Motivation
+
+In order to correlate log entries with the tracing spans that were occurring 
+when the log entry was created, we need to emit certain information in the logs 
+to correlate the data entries. This document attempts to lay out the information 
+that needs to be emitted, and sets forth several approaches for doing so.
+
+## Explanation
+
+Two types of correlation need to happen to tie a log record into its full 
+execution context. The first is Request Correlation, which ties the record to 
+operations that were occurring when the record was created. The second is 
+Resource Correlation, which associates the entry with where the event occurred 
+such as a host, a pod, or a virtual machine.
+
+### Request Correlation
+
+Trace correlation is achieved primarily with two values- a trace identifier and 
+a span identifier. A trace may contain multiple spans, arranged as a tree, and 
+may also contain links to other related spans. The combination of a trace 
+identifier and a span identifier corresponds to a specific scope of work. In the 
+tracing API, that scope can also contain attributes and events that describe 
+what work the program being traced was currently performing.
+
+In most cases when traces and logs are used in tandem, the attributes of the 
+current scope do not need to be added to the log entry- doing so would duplicate 
+transmission of those values. As one span is likely associated with several (or 
+many) log entries, it is more efficient to transmit those attributes with the 
+span once rather than many times with each log entry. As a result, for most 
+purposes the goal is to set the three values of traceid, spanid, and traceflags 
+into each log entry that should be associated with that span.
+
+| Field      | Required | Format  
+| :--------- | :------- | :--------------------------------------------------
+| traceid    | Yes      | 16-byte numeric value as Base16-encoded string
```

Comment:
This refers to base-16 encoded string. However, what if my logging medium is binary and allows proper representation of arbitrary by sequences? Do I still have to use base-16 strings or I can just emit the bytes? If this recommendation is for text medium (such as text log files) it may be worth calling out specifically.

Reply:
I'll clarify, thanks.